src/ivan_cherednikov_ex/ex05/myrand.py

```diff
@@ -0,0 +1,97 @@
+# -*- coding: utf-8 -*-
+
+__author__ = 'Ivan Cherednikov'
+__email__ = 'ivch@nmbu.no'
+
+
+class LCGRand:
+
+    def __init__(self, seed):
+        self.seed = seed
+        self.a = 7 ** 5
+        self.m = 2 ** 31 - 1
+
+    def rand(self):
+        """
+        Calculates next random number, LCG method
+        """
+        self.seed = self.a * self.seed % self.m
+        return self.seed
+
+    def random_sequence(self, length):
+        return RandIter(self, length)
+
+    def infinite_random_sequence(self):
+        while True:
+            yield self.rand()
+
+
+class RandIter:
+    def __init__(self, random_number_generator, length):
+        """
+        Arguments
+        ---------
+        random_number_generator :
+            A random number generator with a ``rand`` method that
+            takes no arguments and returns a random number.
+        length : int
+            The number of random numbers to generate
+        """
+        self.generator = random_number_generator
+        self.length = length
+        self.num_generated_numbers = None
+
+    def __iter__(self):
+        """
+        Initialise the iterator.
+
+        Returns
+        -------
+        self : RandIter
+
+        Raises
+        ------
+        RuntimeError
+            If iter is called twice on the same RandIter object.
+        """
+        if self.num_generated_numbers is not None:
+            raise RuntimeError()
```

Comment:
Maybe put some message in here so we know what went wrong when it raises the error.